src/Core/Resolvers/Factories/QueryEngineFactory.cs

```diff
@@ -49,10 +84,15 @@ public QueryEngineFactory(RuntimeConfigProvider runtimeConfigProvider,
 
             if (config.CosmosDataSourceUsed)
             {
-                IQueryEngine queryEngine = new CosmosQueryEngine(cosmosClientProvider, metadataProviderFactory, authorizationResolver, gQLFilterParser, runtimeConfigProvider, cache);
+                IQueryEngine queryEngine = new CosmosQueryEngine(_cosmosClientProvider, _metadataProviderFactory, _authorizationResolver, _gQLFilterParser, _runtimeConfigProvider, _cache);
                 _queryEngines.Add(DatabaseType.CosmosDB_NoSQL, queryEngine);
             }
+        }
 
+        public void QueryEngineFactoryOnConfigChanged(object? sender, HotReloadEventArgs args)
```

Comment:
because you have the separate `DabConfigEvents` constants, do these service method names need to be prefixed with the enclosing class name still? 
e.g. `QueryEngineFactory::QueryEngineFactoryOnConfigChanged` -> `QueryEngineFactory::OnConfigChanged`

Reply:
Can simplify the names for most, a few still need them to distinguish between base and child classes. I updated to reflect this.